lp-build-core.py

```diff
@@ -0,0 +1,146 @@
+        self.start_build_time = None
+
+        # we need to store credentials once for cronned builds
+        self.cachedir = os.path.join(workdir, 'cache')
+        self.creds = os.path.join(workdir, 'credentials')
+
+    def build(self):
+        """ Trigger the building system for the core snap in the defined arches """
+        launchpad = Launchpad.login_with('Ubuntu Core Builds',
+                                         'production', self.cachedir,
+                                         credentials_file=self.creds,
+                                         version='devel')
+
+        # get snappy-dev team data and ppa
+        snappydev = launchpad.people[self.people]
+        imageppa = snappydev.getPPAByName(name=self.ppa)
+
+        # get snap
+        self.ubuntucore = launchpad.snaps.getByName(name=self.snap,
+                                               owner=snappydev)
+
+        # get distro info
+        ubuntu = launchpad.distributions[self.distribution]
+        release = ubuntu.getSeries(name_or_version=self.series)
+
+        # print a stamp
+        self.start_build_time = datetime.now()
+        print('Trying to trigger builds at: {}'.format(
+            self.start_build_time.strftime(self.stamp_format)))
+
+        # loop over arches and trigger builds
+        for buildarch in self.arches:
+            arch = release.getDistroArchSeries(archtag=buildarch)
+            request = self.ubuntucore.requestBuild(archive=imageppa,
+                                              distro_arch_series=arch,
+                                              pocket='Proposed')
+            buildid = str(request).rsplit('/', 1)[-1]
+            self.builds.append(buildid)
+            print('Arch: {} is building under: {}'.format(buildarch, request))
+
+    def check_builds(self, timeout):
+        """ check the status each minute until all builds have finished
+        :param timeout: used to stop the checks once the timeout it reached
+        """
+
+        print('Receiving status for builds')
+        while len(self.builds):
+            elapsed_build_time = int((datetime.now() - self.start_build_time).total_seconds())
+            if timeout < elapsed_build_time:
+                print('Timeout reached')
+                exit(1)
+            else:
+                print('Remaining {} seconds to timeout'.format(timeout - elapsed_build_time))
+
+            for build in self.builds:
+                try:
+                    response = self.ubuntucore.getBuildSummariesForSnapBuildIds(snap_build_ids=[build])
+                except Exception as e:
+                    print('Could not get response for {}, error: {})'.format(build, e))
+                    continue
+                status = response[build]['status']
+                print('Received response for build {} with status: {}'.format(build, status))
+
+                if status == 'FULLYBUILT':
+                    self.builds.remove(build)
+                    continue
```

Comment:
The `continue` statements in each of the branches here are no-ops since this is the last statement in the loop body.